bin/range-finder/src/cli.rs

```diff
@@ -0,0 +1,242 @@
+//! CLI for the range-finder.
+
+use clap::{ArgAction, Parser};
+use color_eyre::eyre::{eyre, Result};
+use kona_derive::{
+    online::*,
+    types::{L2BlockInfo, StageError},
+};
+use reqwest::Url;
+use std::sync::Arc;
+use superchain_registry::ROLLUP_CONFIGS;
+use tracing::Level;
+use tracing::{debug, error, trace, warn};
+
+const LOG_TARGET: &str = "range_finder";
+
+/// Range Finder Cli
+///
+/// The CLI struct needs a few RPC URLs that it uses as inputs
+/// to the derivation pipeline to derive the range of L2 blocks.
+#[derive(Parser, Clone, Debug)]
+#[command(author, version, about, long_about = None)]
+pub struct Cli {
+    /// Verbosity level (0-4)
+    #[arg(long, short, help = "Verbosity level (0-4)", action = ArgAction::Count)]
+    pub v: u8,
+    /// Starting L2 Block
+    #[arg(long, short, help = "Starting L2 Block")]
+    pub start_block: u64,
+    /// Ending L2 Block
+    #[arg(long, short, help = "Ending L2 Block")]
+    pub end_block: u64,
+    /// The L1 PRC url  for fetching L1 block info.
+    #[arg(long, short, help = "The L1 PRC url for fetching L1 block info.")]
+    pub l1_rpc_url: String,
+    /// An L2 RPC URL to fetch L2 block data from.
+    #[clap(long, help = "RPC url to fetch L2 block data from")]
+    pub l2_rpc_url: String,
+    /// A beacon url for fetching blob information.
+    #[arg(long, short, help = "A beacon url for fetching blob information.")]
+    pub beacon_url: String,
+}
+
+impl Cli {
+    /// Initializes telemtry for the application.
+    pub fn init_telemetry(self) -> Result<Self> {
+        color_eyre::install()?;
+        let subscriber = tracing_subscriber::fmt()
+            .with_max_level(match self.v {
+                0 => Level::ERROR,
+                1 => Level::WARN,
+                2 => Level::INFO,
+                3 => Level::DEBUG,
+                _ => Level::TRACE,
+            })
+            .finish();
+        tracing::subscriber::set_global_default(subscriber).map_err(|e| eyre!(e))?;
+        Ok(self)
+    }
+
+    /// Parse the CLI arguments and run the command
+    pub async fn run(&self) -> Result<()> {
+        // Build the pipeline
+        let cfg = Arc::new(self.rollup_config().await?);
+        let mut l1_provider = self.l1_provider()?;
+        let mut l2_provider = self.l2_provider(cfg.clone())?;
+        let attributes = self.attributes(cfg.clone(), &l2_provider, &l1_provider);
+        let blob_provider = self.blob_provider();
+        let dap = self.dap(l1_provider.clone(), blob_provider, &cfg);
+        let mut l2_cursor = self.cursor().await?;
+        let l1_tip = l1_provider
+            .block_info_by_number(l2_cursor.l1_origin.number)
+            .await
+            .expect("Failed to fetch genesis L1 block info for pipeline tip");
+        let mut pipeline = new_online_pipeline(
+            cfg,
+            l1_provider.clone(),
```

Comment:
You shouldn't need this clone. Since you are passing a reference to `l1_provider` above